jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/PersistenceSuite.scala

```diff
@@ -124,6 +98,8 @@ class PersistenceSuite extends FunSuite with PerTest with BeforeAndAfterAll {
   }
 
   test("test persistence of MLlib pipeline with XGBoostClassificationModel") {
+    val tempDir = createTmpFolder("MLLib").toFile
```

Comment:
why we need to create tempDir in each test manually?

Reply:
As far as I know in scalatest different tests are not isolated.
I create a temp folder once per suite. So one need to create temp folders in each test explicitly to make them isolated.

Alternative solution is to change a trait to create a temp folder per test

Reply:
but what's the problem with reusing the created dir here?

Reply:
I would prefer keeping writing test easy by forcing devs to remember call createTmpFolder manually,

if reusing the directory doesn't bring any issue, can we just reuse the created one in beforeAll()?

Reply:
@trams would you please address this so that we can merge it?